app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -102,11 +108,11 @@ def alternate_machine_id(machine_id)
         # business knowledge.
         # 1 - https://bugzilla.redhat.com/show_bug.cgi?id=1294461
         alternate = []
-        alternate << swap_part(machine_id[0, 8])
-        alternate << swap_part(machine_id[8, 4])
-        alternate << swap_part(machine_id[12, 4])
-        alternate << machine_id[16, 4]
-        alternate << machine_id[20, 12]
+        alternate << swap_part(machine_id[0, 8]) unless machine_id[0, 8].nil?
```

Comment:
Isn't there a way to have those nil checks around the whole block? Or can parts of the `machine_id` be nil and others not?